api/app_sph_lms/api/view/trainer_view.py

```diff
@@ -15,3 +15,18 @@ def list(self, request, *args, **kwargs):
 
     def get_queryset(self):
         return Course.objects.filter(author=self.request.user)
+
+class TrainerLearningPathList(generics.ListCreateAPIView):
+    queryset = LearningPath.objects.filter(is_active=True)
+    serializer_class = TrainerLearningPathSerializer
+    pagination_class = LargeResultsSetPagination
+
+    def get(self, request, *args, **kwargs):
+        self.user_id = request.user.id
+        return super().get(request, *args, **kwargs)
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        queryset = queryset.filter(author=self.user_id)
+
+        return queryset
```

Comment:
You can remove the `get` method, and assign user ID directly.

```suggestion
    def get_queryset(self):
        queryset = super().get_queryset()
        queryset = queryset.filter(author=self.request.user.id)   
        # or try self.request.user if it works

        return queryset
```